### PR TITLE
ztp: Update container builder to golang 1.17

### PR DESCRIPTION
**Pull request by lack**
Signed-off-by: Jim Ramsay <jramsay@redhat.com>

### CODE REVIEW

File: ztp/resource-generator/Containerfile [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # Builder
-FROM openshift/origin-release:golang-1.16 as builder
+FROM registry.ci.openshift.org/openshift/release:golang-1.17 as builder
```
Comment (nishant-parekh) [inline ztp/resource-generator/Containerfile:2]: any particular reason we want to use the ci registry instead of a catalog released build?

Comment (lack) [reply ztp/resource-generator/Containerfile:2]: It's the same image that cnf-tests uses, no other reason.

Comment (lack) [reply ztp/resource-generator/Containerfile:2]: Also to use the released build I'd have to make more changes in the CI lane so it builds.

Comment (nishant-parekh) [reply ztp/resource-generator/Containerfile:2]: makes sense!

**Pull request merged**